src/lib/boavizta/index.ts

```diff
@@ -164,219 +61,178 @@ export class BoaviztaCpuOutputModel
+   */
+  private async validateProvider(staticParamsCast: ModelParams) {
+    const supportedProviders =
+      await this.boaviztaAPI.getSupportedProvidersList();
+
+    if (!supportedProviders.includes(staticParamsCast.provider)) {
       throw new InputValidationError(
         this.errorBuilder({
-          message: 'Missing configuration parameters',
+          message: `Invalid 'provider' parameter '${
+            staticParamsCast.provider
+          }'. Valid values are ${supportedProviders.join(', ')}`,
         })
       );
     }
+  }
 
-    const dataCast = this.sharedParams as KeyValuePair;
+  /**
+   * Validates the instance type parameter for the Cloud model.
+   */
+  private async validateInstanceType(staticParamsCast: ModelParams) {
+    const provider = staticParamsCast.provider;
 
-    for (const key in dataCast) {
-      // replace - with _ in keys
-      if (key.includes('-')) {
-        const newKey = key.replace(/-/g, '_');
-        dataCast[newKey] = dataCast[key];
-        delete dataCast[key];
-      }
+    if (
+      !this.instanceTypes[provider] ||
+      this.instanceTypes[provider].length === 0
+    ) {
+      this.instanceTypes[provider] =
+        await this.boaviztaAPI.getSupportedInstancesList(provider);
     }
 
-    dataCast['usage'] = usageData;
-    const response = await axios.post(
-      `https://api.boavizta.org/v1/cloud/instance?verbose=${this.verbose}&duration=${dataCast['usage']['hours_use_time']}`,
-      dataCast
-    );
-    return this.formatResponse(response.data);
-  }
-
-  protected async captureStaticParams(staticParams: object): Promise<object> {
     if (
-      'verbose' in staticParams &&
-      staticParams.verbose !== undefined &&
-      (staticParams.verbose === true || staticParams.verbose === false)
+      !this.instanceTypes[provider].includes(staticParamsCast['instance-type'])
     ) {
-      this.verbose = staticParams.verbose;
-      staticParams.verbose = undefined;
+      throw new UnsupportedValueError(
+        this.errorBuilder({
+          message: `Invalid 'instance-type' parameter: '${
+            staticParamsCast['instance-type']
+          }'. Valid values are : ${this.instanceTypes[provider].join(', ')}`,
+        })
+      );
     }
+  }
 
-    await this.validateProvider(staticParams); // if no valid provider found, throw error
-
-    await this.validateInstanceType(staticParams); // if no valid 'instance-type' found, throw error
+  /**
+   * Validates the location parameter for the Cloud model.
+   */
+  private async validateLocation(
+    staticParamsCast: ModelParams
+  ): Promise<string | void> {
+    if ('location' in staticParamsCast) {
+      const location = staticParamsCast.location as string;
+      const countries = await this.boaviztaAPI.getSupportedLocations();
 
-    await this.validateLocation(staticParams); // if no valid location found, throw error
+      if (!countries.includes(location)) {
+        throw new InputValidationError(
```

Comment:
can be like this to make error message lighter
```
const whitelistedCountries = countries.join(', ');

        throw new InputValidationError(
          this.errorBuilder({
            message: `Invalid location parameter location. Valid values are ${whitelistedCountries}`,
          })
        );
```

Reply:
done